### ISSUE TITLE
sphinxcontrib/spelling/tests/test_filter.py::test_contributors fails when not run from git checkout

### ISSUE THREAD

**Issue by mgorny** [sphinx-contrib/spelling#70]
When running tests from pypi archive or GitHub snapshot, `test_contributors` fails:

```
$ tox -e py38
py38 create: /tmp/test/sphinxcontrib-spelling-5.4.0/.tox/py38
py38 installdeps: .[test]
py38 develop-inst: /tmp/test/sphinxcontrib-spelling-5.4.0
py38 installed: alabaster==0.7.12,attrs==20.2.0,Babel==2.8.0,certifi==2020.6.20,chardet==3.0.4,coverage==5.2.1,docutils==0.16,extras==1.0.0,fixtures==3.0.0,idna==2.10,imagesize==1.2.0,iniconfig==1.0.1,Jinja2==2.11.2,linecache2==1.0.0,MarkupSafe==1.1.1,more-itertools==8.5.0,packaging==20.4,pbr==5.5.0,pluggy==0.13.1,py==1.9.0,pyenchant==3.1.1,Pygments==2.6.1,pyparsing==2.4.7,pytest==6.0.1,pytest-cov==2.10.1,python-mimeparse==1.6.0,pytz==2020.1,requests==2.24.0,six==1.15.0,snowballstemmer==2.0.0,Sphinx==3.2.1,sphinxcontrib-applehelp==1.0.2,sphinxcontrib-devhelp==1.0.2,sphinxcontrib-htmlhelp==1.0.3,sphinxcontrib-jsmath==1.0.1,sphinxcontrib-qthelp==1.0.3,sphinxcontrib-serializinghtml==1.1.4,# Editable install with no version control (sphinxcontrib-spelling==5.4.0),-e /tmp/test/sphinxcontrib-spelling-5.4.0,testtools==2.4.0,toml==0.10.1,traceback2==1.4.0,unittest2==1.1.0,urllib3==1.25.10
py38 run-test-pre: PYTHONHASHSEED='174003223'
py38 run-test: commands[0] | pytest -v --cov=sphinxcontrib.spelling --cov-report term-missing --cov-config /tmp/test/sphinxcontrib-spelling-5.4.0/.coveragerc
=============================================================== test session starts ===============================================================
platform linux -- Python 3.8.5, pytest-6.0.1, py-1.9.0, pluggy-0.13.1 -- /tmp/test/sphinxcontrib-spelling-5.4.0/.tox/py38/bin/python
cachedir: .tox/py38/.pytest_cache
rootdir: /tmp/test/sphinxcontrib-spelling-5.4.0
plugins: cov-2.10.1
collected 17 items                                                                                                                                

sphinxcontrib/spelling/tests/test_builder.py::test_setup PASSED                                                                             [  5%]
sphinxcontrib/spelling/tests/test_builder.py::test_title PASSED                                                                             [ 11%]
sphinxcontrib/spelling/tests/test_builder.py::test_body PASSED                                                                              [ 17%]
sphinxcontrib/spelling/tests/test_builder.py::test_ignore_literals PASSED                                                                   [ 23%]
sphinxcontrib/spelling/tests/test_builder.py::test_several_word_lists PASSED                                                                [ 29%]
sphinxcontrib/spelling/tests/test_builder.py::test_ignore_file PASSED                                                                       [ 35%]
sphinxcontrib/spelling/tests/test_checker.py::test_errors_only PASSED                                                                       [ 41%]
sphinxcontrib/spelling/tests/test_checker.py::test_with_suggestions PASSED                                                                  [ 47%]
sphinxcontrib/spelling/tests/test_checker.py::test_with_wordlist PASSED                                                                     [ 52%]
sphinxcontrib/spelling/tests/test_checker.py::test_with_context_line PASSED                                                                 [ 58%]
sphinxcontrib/spelling/tests/test_checker.py::test_line_of_index_one_line PASSED                                                            [ 64%]
sphinxcontrib/spelling/tests/test_checker.py::test_line_of_index_multi_line PASSED                                                          [ 70%]
sphinxcontrib/spelling/tests/test_filter.py::test_builtin_unicode PASSED                                                                    [ 76%]
sphinxcontrib/spelling/tests/test_filter.py::test_builtin_regular PASSED                                                                    [ 82%]
sphinxcontrib/spelling/tests/test_filter.py::test_acronym PASSED                                                                            [ 88%]
sphinxcontrib/spelling/tests/test_filter.py::test_acronym_unicode PASSED                                                                    [ 94%]
sphinxcontrib/spelling/tests/test_filter.py::test_contributors FAILED                                                                       [100%]

==================================================================== FAILURES =====================================================================
________________________________________________________________ test_contributors ________________________________________________________________

    def test_contributors():
        f = filters.ContributorFilter(None)
        names = [
            "Alex",
            "Atlakson",
            "Avram",
            "Baumgold",
            "Berman",
            "Daniele",
            "Doug",
            "Finucane",
            "Gaynor",
            "Gonsiorowski",
            "Hong",
            "Hong",
            "Huon",
            "Kampik",
            "Kolosov",
            "Lubkin",
            "Marti",
            "Minhee",
            "Olausson",
            "Raggam",
            "Raudsepp",
            "sdelliot",
            "Sergey",
            "Sevilla",
            "Timotheus",
            "Tobias",
            "Tricoli",
        ]
        for name in names:
>           assert f._skip(name)
E           AssertionError: assert False
E            +  where False = <bound method IgnoreWordsFilter._skip of <sphinxcontrib.spelling.filters.ContributorFilter object at 0x7f5be9dd2460>>('Alex')
E            +    where <bound method IgnoreWordsFilter._skip of <sphinxcontrib.spelling.filters.ContributorFilter object at 0x7f5be9dd2460>> = <sphinxcontrib.spelling.filters.ContributorFilter object at 0x7f5be9dd2460>._skip

sphinxcontrib/spelling/tests/test_filter.py:71: AssertionError
-------------------------------------------------------------- Captured stderr call ---------------------------------------------------------------
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
================================================================ warnings summary =================================================================
sphinxcontrib/spelling/filters.py:11
  /tmp/test/sphinxcontrib-spelling-5.4.0/sphinxcontrib/spelling/filters.py:11: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

sphinxcontrib/spelling/tests/test_builder.py::test_setup
sphinxcontrib/spelling/tests/test_builder.py::test_title
sphinxcontrib/spelling/tests/test_builder.py::test_body
sphinxcontrib/spelling/tests/test_builder.py::test_ignore_literals
sphinxcontrib/spelling/tests/test_builder.py::test_several_word_lists
sphinxcontrib/spelling/tests/test_builder.py::test_ignore_file
  /tmp/test/sphinxcontrib-spelling-5.4.0/sphinxcontrib/spelling/checker.py:34: DeprecationWarning: passing 'filters' as a non-keyword argument to get_tokenizer() is deprecated
    self.tokenizer = get_tokenizer(tokenizer_lang, filters)

-- Docs: https://docs.pytest.org/en/stable/warnings.html

----------- coverage: platform linux, python 3.8.5-final-0 -----------
Name                                  Stmts   Miss  Cover   Missing
-------------------------------------------------------------------
sphinxcontrib/spelling/__init__.py       32      3    91%   6-8, 23
sphinxcontrib/spelling/asset.py          20      7    65%   23-30
sphinxcontrib/spelling/builder.py       144     18    88%   23-24, 44, 68-69, 82, 101-106, 145, 156, 161, 195-196, 214
sphinxcontrib/spelling/checker.py        44      4    91%   11-12, 27, 44
sphinxcontrib/spelling/directive.py      21     12    43%   30-48
sphinxcontrib/spelling/filters.py        85     19    78%   38-39, 42-45, 129, 133, 137, 157, 160, 167-168, 177-178, 195, 231-233
-------------------------------------------------------------------
TOTAL                                   346     63    82%

============================================================= short test summary info =============================================================
FAILED sphinxcontrib/spelling/tests/test_filter.py::test_contributors - AssertionError: assert False
==================================================== 1 failed, 16 passed, 7 warnings in 2.11s =====================================================
ERROR: InvocationError for command /tmp/test/sphinxcontrib-spelling-5.4.0/.tox/py38/bin/pytest -v --cov=sphinxcontrib.spelling --cov-report term-missing --cov-config .coveragerc (exited with code 1)
_____________________________________________________________________ summary _____________________________________________________________________
ERROR:   py38: commands failed
```

Please fix it to work on data included in the tarball. Using git checkout does not really work for distribution packagers.

**Comment by dhellmann** [comment]
That filter works based on git data. Any variation of the test to make it use fake data wouldn't really test the filter code.

Maybe we can add a way for you to skip that test? Do you have any similar cases that we could use as a basis for that? Would an environment variable work, for example?

**Comment by mgorny** [comment]
I suppose the simplest solution would be to check for `.git`.

**Comment by dhellmann** [comment]
> I suppose the simplest solution would be to check for `.git`.

That could work.

**Comment by maglo** [comment]
I agree that we should fail silently if it is not a Git repository. We have enabled "fail on warn" in Sphinx to fail on Warnings, and this plugin to Warn. So that documents that are not spelled correctly will not pass build steps. For sure, all our CI are running from a Git repo.

But anyway, if somebody wants to run this plugin without a Git repo, we should fail silently and just not add Git information.

Great work with this plugin, btw. ❤️

**Comment by dhellmann** [comment]
I'll be happy to review a patch to make it possible for someone to run the test suite and skip that test explicitly, have the test skip itself if there is no git repo, or have the test create its own git repo for testing.

**Comment by dhellmann** [comment]
This was fixed by b1680cdb
